File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> set=new HashSet<>();
        int count=0,m=0,j=0;
        if(s.length()==0 || s.length()==1){
            return s.length();
        }
        for(int i=0;i<s.length();i++){
            while(j<i && set.contains(s.charAt(i))){
                set.remove(s.charAt(j++));
            }
            set.add(s.charAt(i));
                m=Math.max(m,set.size());
        }
        return m;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        st=set()
        i=j=m=0
        while j<len(s):
            while s[j] in st and i<=j:
                st.remove(s[i])
                i+=1
            st.add(s[j])
            j+=1
            m=max(m,len(st))
        return m

        